src/main/kotlin/com/mdrobnak/lalrpop/psi/ext/LpAction.kt

```diff
@@ -81,51 +81,53 @@ fun LpAction.actionCodeFunctionHeader(withReturnType: Boolean = true): String {
         grammarWhereClauses?.grammarWhereClauseList?.joinToString(prefix = "where ", separator = ", ") { it.text }
             ?: ""
 
-    return "fn __intellij_lalrpop $genericParamsString ($grammarParametersString $arguments) $arrowReturnType\n" +
-            "$grammarWhereClausesString\n"
+    return """
+        fn __intellij_lalrpop $genericParamsString ($grammarParametersString $arguments) $arrowReturnType
+        $grammarWhereClausesString
+        """.trimIndent()
 }
 
 abstract class LpActionMixin(node: ASTNode) : ASTWrapperPsiElement(node), LpAction {
     override fun isValidHost(): Boolean = true
 
-    override fun updateText(text: String): PsiLanguageInjectionHost {
+    override fun updateText(text: String): PsiLanguageInjectionHost = apply {
         val valueNode = node.lastChildNode
         assert(valueNode is LeafElement)
         (valueNode as LeafElement).replaceWithText(text)
-        return this
     }
```

Comment:
It was only to avoid `return this`

Reply:
Right, though before the intention to return `this` was clear. Now with `apply`, that intention is less clear as the end of the method does not seem to explicitly return anything (looks the same as if it was a normal `fun foo() { ... }`. Basically, I think using `apply` in this way is a little too "clever" such that it hurts readability.

Reply:
Fixed in 867d6d5

Reply:
Missed one

Reply:
Also what about those with `element.apply` ([LpNonterminalReference.kt:18](https://github.com/Mcat12/intellij-lalrpop/pull/32/files#diff-1190c0866365e1f58d2afe0e076c11dfbda6a7f7511299e5bc210c6cc7c313d5R18), [LpPathIdReference.kt:18](https://github.com/Mcat12/intellij-lalrpop/pull/32/files#diff-0d6be29c18c87715066e2e66ca6bb4f91163ecf4f1d165dbac5d0802ab9f0b21R18)). Should I change them too?

Reply:
Those are slightly better usages since they're closer to what I expect `apply` to be used for (ex. [here](https://github.com/yole/idiomatic-kotlin/blob/master/src/main/kotlin/p03_stdlib/t18_Apply/good.kt)), so I think they're OK to have. I also noted this in https://github.com/Mcat12/intellij-lalrpop/pull/32#discussion_r553462664 (but you may have missed it since I resolved it early in the review).